### ISSUE TITLE
check REST api docs

### ISSUE THREAD

**Issue by ThomasWaldmann** [bepasty/bepasty-server#249]
https://github.com/bepasty/bepasty-server/blob/master/docs/source/rest.rst

are they uptodate / complete?

@OGAWAHirofumi can you have a look?

**Comment by OGAWAHirofumi** [comment]
missing new REST apis. 
```
POST '/apis/rest/items/<itemname>/delete'
POST '/apis/rest/items/<itemname>/modify'
POST '/apis/rest/items/<itemname>/lock'
POST '/apis/rest/items/<itemname>/unlock'
```

not sure if error response should be documented or not though (404 may not be handled for now, at least),
now, error response from REST api will be by JSON format (was previously partial html or plain string).

```
{
    "error": {
        "code": <http status code>,
        "message": <error detail>
    }
}
```

[delete]
delete a item specified by `<itemname>`

params:
no parameter

return value:
if success, status 200. otherwise status != 200

[modify]
modify metadatas of a item specified by `<itemname>`

params:
new metadata is specified by JSON. now support 'filename' and 'type'
for example, if you want to modify filename
```
{"filename": "new-filename.txt"}
```
when you modify both of filename and type,
```
{"filename": "new-filename.txt", "type": "new-mimetype"}
```

return value:
if success, status 200. otherwise status != 200

[lock]
lock a item specified by `<itemname>`

params:
no parameter

return value:
if success, status 200. otherwise status != 200

[unlock]
unlock a item specified by `<itemname>`

params:
no parameter

return value:
if success, status 200. otherwise status != 200

**Comment by ThomasWaldmann** [comment]
Can you do a PR for the docs?

You can test them locally with `cd docs ; make html` and then point your browser at `index.html` in there.

**Comment by ThomasWaldmann** [comment]
fixed by #251

**Comment by ThomasWaldmann** [comment]
@OGAWAHirofumi btw, you can automatically close such tickets as this one by writing "..., fixes #249" in the commit comment of the fix. When that is merged, github will close the ticket.
